src/domain/app/AppRoute.tsx

```diff
@@ -25,30 +22,38 @@ function AppRoute({
   //
   // For instance, the useAuthenticated hook replaces the current path,
   // in which case the original navigation target would be lost.
-  useInstantLogin(isPrivate);
-  const canRenderComponent = useAuthenticated(isPrivate);
+  const { isAuthenticated } = useOidcClient();
+
+  const canRenderComponent = !isPrivate || isAuthenticated();
+
+  const navigate = useNavigate();
   const { t } = useTranslation();
 
   if (!title && !noTitle) {
     // eslint-disable-next-line no-console
     console.warn(
       // eslint-disable-next-line max-len
-      'Most app routes should have a title. If a route does not need one, explicitly state is by toggling the noTitle prop.'
+      'Most app routes should have a title. If a route does not need one, explicitly state is by toggling the noTitle prop.',
+      { element }
     );
   }
 
+  if (!canRenderComponent) {
+    navigate('/unauthorized', { replace: true });
+  }
+
   return (
     <>
       {/* eslint-disable-next-line @typescript-eslint/ban-ts-comment */}
       {/* @ts-ignore ts2786 - https://github.com/nfl/react-helmet/issues/646 */}
       <Helmet>
         {title && (
           <title>
-            {title} - {t('appName')}
+            {t(title)} - {t('appName')}
```

Comment:
Renaming the `title` to `titleKey` and adding `const title = t(titleKey)` right after useTranslation call in tandem with the added documentation to AppRouteProps might be even clearer, and it would handle the truish checks for `title` like it was before.

Reply:
The earlier implementation was that the translation was already done in the router, but it didn't do anything 😞 The translation was needed to be done later on.

I will change the property name.